runtime/stdlib/account.go

```diff
@@ -1595,31 +1596,33 @@ func changeAccountContracts(
 		memoryGauge := invocation.Interpreter.SharedState.Config.MemoryGauge
 		legacyUpgradeEnabled := invocation.Interpreter.SharedState.Config.LegacyContractUpgradeEnabled
 
-		oldProgram, err := parser.ParseProgram(
-			memoryGauge,
-			oldCode,
-			parser.Config{
-				IgnoreLeadingIdentifierEnabled: true,
-			},
-		)
+		var oldProgram *ast.Program
 
-		var legacyContractUpgrade bool
-		// if we are allowing legacy contract upgrades, fall back to the old parser when the new one fails
-		if !ignoreUpdatedProgramParserError(err) && legacyUpgradeEnabled {
-			legacyContractUpgrade = true
+		// It is not always possible to determine whether the old code is pre-1.0 or not,
+		// only based on the parser errors. Therefore, always rely on the flag only.
+		// If the legacy contract upgrades are enabled, then use the old parser.
+		if legacyUpgradeEnabled {
```

Comment:
This is an unrelated change, but realized that the existing condition here used  to rely on parser errors to determine whether this is a legacy-code update. 
However, not all pre-1.0 codes may have parser errors (majority of them would have checking errors, but probably not parser errors always). Therefore, make this simpler, and always use the config flag to determine whether this is a legacy-code update or not. Developers can restart the emulator with/without flag to switch between the use-cases.

Reply:
Good catch! Agreed